src/main/java/tuteez/logic/commands/AddCommand.java

```diff
@@ -81,37 +81,71 @@ public CommandResult execute(Model model) throws CommandException {
-        for (Lesson lesson: lessonLst) {
-            assert lesson != null;
-            Map<Person, ArrayList<Lesson>> clashingLessons = model.getClashingLessons(lesson);
-
-            clashingLessons.forEach((person, lessons) -> {
-                assert !person.equals(toAdd) : "By this point a duplicate person should be flagged out";
-                resultMap.computeIfAbsent(person, k -> new ArrayList<>()).addAll(lessons);
-            });
-        }
+        Map<Person, ArrayList<Lesson>> resultMap = findAllClashingLessonsMap(model, lessonLst);
 
         if (!resultMap.isEmpty()) {
             String logMessage = String.format("Student: %s | Lessons: %s | Conflict: Clashes with "
                     + "another student's lesson", toAdd.getName(), toAdd.getLessons().toString());
             logger.info(logMessage);
 
-            StringBuilder clashMsg = new StringBuilder(MESSAGE_CLASHING_LESSON).append("\n");
-            resultMap.keySet().forEach(student -> {
-                clashMsg.append(student.getName()).append(": ");
-
-                resultMap.get(student).forEach(ls -> clashMsg.append(ls.getDayAndTime()).append(" "));
-                clashMsg.append("\n");
-            });
-            throw new CommandException(clashMsg.toString());
+            String clashMsg = generateClashMessage(resultMap);
+            throw new CommandException(clashMsg);
         }
 
         model.addPerson(toAdd);
         logger.info("Student Added - " + toAdd);
         return new CommandResult(String.format(MESSAGE_SUCCESS, Messages.format(toAdd)));
     }
 
+    /**
+     * Finds all lessons that clash with the provided list of lessons for different students.
+     *
+     * <p>This method checks each lesson in the given list against the existing lessons in the model
+     * to identify any scheduling conflicts with lessons for other students. If any clashes are found,
+     * they are added to a map with the student and their conflicting lessons.</p>
+     *
+     * @param model The model, representing in-memory model for app
+     * @param lessonLst The list of lessons to check for clashes.
+     * @return A map where each key is a {@code Person} representing a student, and the value is an
+     *         {@code ArrayList} of {@code Lesson} objects that are allocated to student and
+     *         conflict with the given lessons.
+     */
+    public Map<Person, ArrayList<Lesson>> findAllClashingLessonsMap(Model model, List<Lesson> lessonLst) {
+        Map<Person, ArrayList<Lesson>> resultMap = new HashMap<>();
+        for (Lesson lesson: lessonLst) {
+            assert lesson != null;
+            Map<Person, ArrayList<Lesson>> clashingLessons = model.getClashingLessons(lesson);
+
+            clashingLessons.forEach((person, lessons) -> {
+                assert !person.equals(toAdd) : "By this point a duplicate person should be flagged out";
+                resultMap.computeIfAbsent(person, k -> new ArrayList<>()).addAll(lessons);
```

Comment:
compute is a vague term

I'll quote the textbook again here

> A name is not just for differentiation; it should explain the named entity to the reader accurately and at a sufficient level of detail.
> `processInput()` (what 'process'?) **BAD** 
> `removeWhiteSpaceFromInput()` **GOOD**

Reply:
replace compute with sth else, then If WHAT Absert

Reply:
`computeIfAbsent` is Java hashmap's API, I did not write that on my own 😅